R/derive_param_tte.R

```diff
@@ -386,6 +390,13 @@ derive_param_tte <- function(dataset = NULL,
   ) %>%
     mutate(!!tmp_event := 1L)
 
+  # check for duplicates in event_data
+  signal_duplicate_records(
+    dataset = event_data,
+    by_vars = expr_c(by_vars, subject_keys),
+    cnd_type = check_type
+  )
```

Comment:
This check will never fail because `filter_date_sources()` returns at most one records per by group and subject. That's the reason why the test "derive_param_tte detects duplicates when check_type = 'warning'" fails.

Reply:
I think I get it. I just made a push, but let me take a look.

Reply:
Okay, let me know if this was effective. The error I was getting was that AEDECOD was not being included in the signal_duplicate_records function. So I bound dataset_adsl and source_datasets together then ran the check within derive_param_tte like so: 

#check for duplicates in dataset_adsl and source_datasets
  combined_dataset <- bind_rows(dataset_adsl, !!!source_datasets)

  signal_duplicate_records(
    dataset = combined_dataset,
    by_vars = expr_c(subject_keys, by_vars),
    cnd_type = check_type
  )

All the tests have passed now with some additional warnings. I have to run to some meetings and seminars today but am pushing now for review and can check back in the evening. On first glance not sure if the earlier tests should get a duplicate warning now as well.